cmd/primarycmd/addValidator.go

```diff
@@ -0,0 +1,202 @@
+	subnetAuthKeys               []string
+	outputTxPath                 string
+	useLedger                    bool
+	ledgerAddresses              []string
+	nodeIDStr                    string
+	weight                       uint64
+	startTimeStr                 string
+	duration                     time.Duration
+	ErrMutuallyExlusiveKeyLedger = errors.New("--key and --ledger,--ledger-addrs are mutually exclusive")
+	ErrStoredKeyOnMainnet        = errors.New("--key is not available for mainnet operations")
+)
+
+type jsonProofOfPossession struct {
+	PublicKey         string `json:"publicKey"`
+	ProofOfPossession string `json:"proofOfPossession"`
+}
+
+// avalanche subnet deploy
+func newAddValidatorCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "addValidator",
+		Short: "Add a validator to Primary Network",
+		Long: `The primary addValidator command adds a node as a validator 
+in the Primary Network`,
+		SilenceUsage: true,
+		RunE:         addValidator,
+		Args:         cobra.ExactArgs(0),
+	}
+	cmd.Flags().StringVarP(&keyName, "key", "k", "", "select the key to use [fuji deploy only]")
+	cmd.Flags().StringVar(&nodeIDStr, "nodeID", "", "set the NodeID of the validator to add")
+	cmd.Flags().Uint64Var(&weight, "weight", 0, "set the staking weight of the validator to add")
+	cmd.Flags().StringVar(&startTimeStr, "start-time", "", "UTC start time when this validator starts validating, in 'YYYY-MM-DD HH:MM:SS' format")
+	cmd.Flags().DurationVar(&duration, "staking-period", 0, "how long this validator will be staking")
+	cmd.Flags().BoolVar(&deployTestnet, "fuji", false, "join on `fuji` (alias for `testnet`)")
+	cmd.Flags().BoolVar(&deployTestnet, "testnet", false, "join on `testnet` (alias for `fuji`)")
+	cmd.Flags().BoolVar(&deployMainnet, "mainnet", false, "join on `mainnet`")
+	cmd.Flags().StringSliceVar(&subnetAuthKeys, "subnet-auth-keys", nil, "control keys that will be used to authenticate add validator tx")
+	cmd.Flags().StringVar(&outputTxPath, "output-tx-path", "", "file path of the add validator tx")
+	cmd.Flags().BoolVarP(&useLedger, "ledger", "g", false, "use ledger instead of key (always true on mainnet, defaults to false on fuji)")
+	cmd.Flags().StringSliceVar(&ledgerAddresses, "ledger-addrs", []string{}, "use the given ledger addresses")
+	return cmd
+}
+
+func promptProofOfPossession() (jsonProofOfPossession, error) {
+	ux.Logger.PrintToUser("Next, we need the public key and proof of possession of the node's BLS")
+	ux.Logger.PrintToUser("SSH into the node and call info.getNodeID API to get the node's BLS info")
+	ux.Logger.PrintToUser("Check https://docs.avax.network/apis/avalanchego/apis/info#infogetnodeid for instructions on calling info.getNodeID API")
+	txt := "What is the public key of the node's BLS?"
+	publicKey, err := app.Prompt.CaptureString(txt)
+	if err != nil {
+		return jsonProofOfPossession{}, err
+	}
+	txt = "What is the proof of possession of the node's BLS?"
+	pop, err := app.Prompt.CaptureString(txt)
```

Comment:
same

Reply:
I found this one creates a lot of lines when pasting, did you saw that. maybe we can solve this also